test/verify/check-examples

```diff
@@ -61,9 +61,9 @@ class TestXHRProxy(MachineCase):
 
         # set up served directory
         httpdir = self.vm_tmpdir + "/root"
-        m.execute("mkdir {0} && echo world > {0}/hello.txt".format(httpdir))
+        m.execute("mkdir {0}; echo world > {0}/hello.txt".format(httpdir))
```

Comment:
why not change this to an f-string?

Reply:
....and possibly also `m.write()`

Reply:
It would mean to repeat `httpdir`, but it's not a very big deal. But let's clean this up once m.write() implies mkdir, then this duplication goes away automatically and f-string is the natural choice.

Reply:
Fixed in PR #17698